script/lazytox.py

```diff
@@ -0,0 +1,234 @@
+
+async def lint(files):
+    """Perform lint."""
+    fres, pres = await asyncio.gather(flake8(files), pylint(files))
+
+    res = fres + pres
+    res.sort(key=lambda item: item.file)
+    if res:
+        print("Pylint & Flake8 errors:")
+    else:
+        printc(PASS, "Pylint and Flake8 passed")
+
+    lint_ok = True
+    for err in res:
+        err_msg = "{} {}:{} {}".format(err.file, err.line, err.col, err.msg)
+
+        # tests/* does not have to pass lint
+        if err.file.startswith('tests/'):
+            print(err_msg)
+        else:
+            printc(FAIL, err_msg)
+            lint_ok = False
+
+    return lint_ok
+
+
+async def main():
+    """The main loop."""
+    # Ensure we are in the homeassistant root
+    os.chdir(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
+
+    files = await git()
+    if not files:
+        print("No changed files found. Please ensure you have added your "
+              "changes with git add & git commit")
+        return
+
+    pyfile = re.compile(r".+\.py$")
+    pyfiles = [file for file in files if pyfile.match(file)]
+
+    print("=============================")
+    printc('bold', "CHANGED FILES:\n", '\n '.join(pyfiles))
+    print("=============================")
+
+    skip_lint = len(sys.argv) > 1 and sys.argv[1] == '--skiplint'
+    if skip_lint:
+        printc(FAIL, "LINT DISABLED")
+    elif not await lint(pyfiles):
+        printc(FAIL, "Please fix your lint issues before continuing")
+        return
+
+    test_files = set()
+    gen_req = False
+    for fname in pyfiles:
+        if fname.startswith('homeassistant/components/'):
+            gen_req = True  # requirements script for components
+        # Find test files...
+        if fname.startswith('tests/'):
+            if '/test_' in fname:  # All test helpers should be excluded
+                test_files.add(fname)
+        else:
+            parts = fname.split('/')
+            parts[0] = 'tests'
+            if parts[-1] == '__init__.py':
+                parts[-1] = 'test_init.py'
+            elif parts[-1] == '__main__.py':
+                parts[-1] = 'test_main.py'
+            else:
+                parts[-1] = 'test_' + parts[-1]
+            fname = '/'.join(parts)
+            if os.path.isfile(fname):
+                test_files.add(fname)
+
+    if gen_req:
+        print("=============================")
+        if validate_requirements_ok():
+            printc(PASS, "script/gen_requirements.py passed")
+        else:
+            printc(FAIL, "Please run script/gen_requirements.py")
+            return
+
+    print("=============================")
+    if not test_files:
+        print("No test files identified, ideally you should run tox")
+        return
+
+    code, _ = await async_exec(
+        'pytest', '-vv', '--force-sugar', '--', *test_files, display=True)
+    print("=============================")
+
+    if code == 0:
+        printc(PASS, "Yay! This will most likely pass tox")
```

Comment:
Suggestion for making the logs more fun 😉 (feel free to ignore this, might even cause issues with other terminals)

```
printc(PASS, "Yay! This will most likely pass tox 🎉")
```

Reply:
This will probably run mostly in bash, otherwise it would have been a good idea! 👍